src/lib/crypto/kimchi_backend/gadgets/ec_group.ml

```diff
@@ -1666,7 +1662,10 @@ let%test_unit "Ec_group.add_full" =
             let open Runner.Impl in
             (* Prepare test public inputs *)
             let curve =
-              Curve_params.to_circuit_constants (module Runner.Impl) curve
+              Curve_params.to_circuit_constants
+                ~use_precomputed_gen_doubles:false
```

Comment:
Interesting optimization

Reply:
If it is true, where does it get them from?

Reply:
They are precomputed and constrained as constants with `Curve_params.to_circuit_constants`

Reply:
They are public parameters loaded into circuit constants.